sdk/language-support/ts/daml-ledger/index.ts

```diff
@@ -174,14 +174,17 @@ function lookupTemplateOrUnknownInterface(
   }
 }
 
+const decodeTemplateId = <I extends string>(): jtv.Decoder<I> =>
+  jtv.string() as jtv.Decoder<I>;
+
 /**
  * Decoder for a [[CreateEvent]].
  */
 const decodeCreateEvent = <T extends object, K, I extends string>(
   template: TemplateOrInterface<T, K, I>,
 ): jtv.Decoder<CreateEvent<T, K, I>> =>
   jtv.object({
-    templateId: jtv.constant(template.templateId),
+    templateId: decodeTemplateId(),
```

Comment:
The create events from the ledger will contain template ids which specify the package by package id. The `template` arg passed in may use a package name, so we cannot expect these two to match exactly. I don't think we really need to match them at all though, as the purpose of this function is just to decode an event, and knowing that this field is represented as a string suffices for the types involved.